plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_aws_config.py

```diff
@@ -0,0 +1,101 @@
+    return field(default_factory=lambda: obj)
+
+
+# This class maps to a Ray cluster config. To get more info on each field, please check:
+# https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/aws/example-full.yaml
+# Note: You cannot add arbitrary field not supported by Ray in this conf.
+@dataclass
+class RayClusterConfig:
+    cluster_name: str = "default"
+    min_workers: int = 0
+    max_workers: int = 1
+    initial_workers: int = 0
+    autoscaling_mode: str = "default"
+    docker: Dict[str, Any] = default_field(
+        {"image": "", "container_name": "", "pull_before_run": True, "run_options": []}
+    )
+    target_utilization_fraction: float = 0.8
+    idle_timeout_minutes: int = 5
+    provider: Dict[str, Any] = default_field(
+        {
+            "type": "aws",
+            "region": "us-west-2",
+            "availability_zone": "us-west-2a, us-west-2b",
+        }
+    )
+    auth: Dict[str, Any] = default_field({"ssh_user": "ubuntu"})
+    head_node: Dict[str, Any] = default_field(
+        {
+            "InstanceType": "m5.large",
+            "ImageId": "ami-008d8ed4bd7dc2485",  # Deep Learning AMI (Ubuntu 18.04) Version 27.0
+        }
+    )
+    worker_nodes: Dict[str, Any] = default_field(
+        {
+            "InstanceType": "m5.large",
+            "ImageId": "ami-008d8ed4bd7dc2485",  # Deep Learning AMI (Ubuntu 18.04) Version 27.0
+            "InstanceMarketOptions": {"MarketType": "spot"},
+        }
+    )
+    file_mounts: Dict[str, str] = default_field({})
+    initialization_commands: List[str] = default_field([])
+    setup_commands: List[str] = default_field(
+        [
+            "conda create -n hydra37 python=3.7 -y",
+            "echo 'export PATH=\"$HOME/anaconda3/envs/hydra37/bin:$PATH\"' >> ~/.bashrc",
+            "pip install -U ray",
+            "pip install -U https://hydra-test-us-west-2.s3-us-west-2.amazonaws.com/hydra_core-1.0.0rc1-py3-none-any.whl",
+            "pip install -U https://hydra-test-us-west-2.s3-us-west-2.amazonaws.com/hydra_ray_launcher-0.1.0-py3-none-any.whl",
+        ]
+    )
+    head_setup_commands: List[str] = default_field(["pip install boto3==1.12.34"])
+    worker_setup_commands: List[str] = default_field([])
+    head_start_ray_commands: List[str] = default_field(
+        [
+            "ray stop",
+            "ulimit -n 65536; ray start --head --redis-port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml",
+        ]
+    )
+    worker_start_ray_commands: List[str] = default_field(
+        [
+            "ray stop",
+            "ulimit -n 65536; ray start --address=$RAY_HEAD_IP:6379 --object-manager-port=8076",
+        ]
+    )
+
+
+@dataclass
+class RayAWSConf:
+    # ray.init(**ray_init_config)
+    ray_init_cfg: Dict[str, Any] = default_field({"address": "auto"})
+
+    # @ray.remote(**ray_remote_cfg)
+    ray_remote_cfg: Dict[str, Any] = default_field({"num_cpus": 1, "num_gpus": 0})
```

Comment:
you can have a dataclass describing those things.
(to get runtime type safety for num_gpus and num_gpus for example).